jni/test/jni_array_test.dart

```diff
@@ -0,0 +1,237 @@
+    array.setRange(0, 3, [false, true, true, true], 1);
+    expect(array[0], true);
+    expect(array[1], true);
+    expect(array[2], true);
+    expect(() {
+      final _ = array[-1];
+    }, throwsRangeError);
+    expect(() {
+      array[-1] = false;
+    }, throwsRangeError);
+    expect(() {
+      array[3] = false;
+    }, throwsRangeError);
+    array.delete();
+  });
+  test("Java char array", () {
+    final array = JniArray(const JniCharType(), 3);
+    expect(array.length, 3);
+    array[0] = 'ح';
+    array[1] = '2';
+    array[2] = '3';
+    expect(array[0], 'ح');
+    expect(array[1], '2');
+    expect(array[2], '3');
+    array.setRange(0, 3, ['4', '5', '6', '7'], 1);
+    expect(array[0], '5');
+    expect(array[1], '6');
+    expect(array[2], '7');
+    expect(() {
+      final _ = array[-1];
+    }, throwsRangeError);
+    expect(() {
+      array[-1] = '4';
+    }, throwsRangeError);
+    expect(() {
+      array[3] = '4';
+    }, throwsRangeError);
+    array.delete();
+  });
+  test("Java byte array", () {
+    final array = JniArray(const JniByteTypeClass(), 3);
+    expect(array.length, 3);
+    array[0] = 1;
+    array[1] = 2;
+    array[2] = 3 + 256 * 5; // truncates the input;
+    expect(array[0], 1);
+    expect(array[1], 2);
+    expect(array[2], 3);
+    array.setRange(0, 3, [4, 5, 6, 7], 1);
+    expect(array[0], 5);
+    expect(array[1], 6);
+    expect(array[2], 7);
+    expect(() {
+      final _ = array[-1];
+    }, throwsRangeError);
+    expect(() {
+      array[-1] = 4;
+    }, throwsRangeError);
+    expect(() {
+      array[3] = 4;
+    }, throwsRangeError);
+    array.delete();
+  });
+  test("Java short array", () {
+    final array = JniArray(const JniShortType(), 3);
+    expect(array.length, 3);
+    array[0] = 1;
+    array[1] = 2;
+    array[2] = 3 + 256 * 256 * 5; // truncates the input
+    expect(array[0], 1);
+    expect(array[1], 2);
+    expect(array[2], 3);
+    array.setRange(0, 3, [4, 5, 6, 7], 1);
+    expect(array[0], 5);
+    expect(array[1], 6);
+    expect(array[2], 7);
+    expect(() {
+      final _ = array[-1];
+    }, throwsRangeError);
+    expect(() {
+      array[-1] = 4;
+    }, throwsRangeError);
+    expect(() {
+      array[3] = 4;
+    }, throwsRangeError);
+    array.delete();
+  });
+  test("Java int array", () {
+    final array = JniArray(const JniIntType(), 3);
+    expect(array.length, 3);
+    array[0] = 1;
+    array[1] = 2;
+    array[2] = 3 + 256 * 256 * 256 * 256 * 5; // truncates the input
+    expect(array[0], 1);
+    expect(array[1], 2);
+    expect(array[2], 3);
+    array.setRange(0, 3, [4, 5, 6, 7], 1);
+    expect(array[0], 5);
+    expect(array[1], 6);
+    expect(array[2], 7);
+    expect(() {
+      final _ = array[-1];
+    }, throwsRangeError);
+    expect(() {
+      array[-1] = 4;
+    }, throwsRangeError);
+    expect(() {
+      array[3] = 4;
+    }, throwsRangeError);
+    array.delete();
+  });
+  const epsFloat = 1e-6;
```

Comment:
nit: we usually write out names

https://dart.dev/guides/language/effective-dart/documentation#avoid-abbreviations-and-acronyms-unless-they-are-obvious